csrc/fusion_profiler.cpp

```diff
@@ -326,17 +326,15 @@ SegmentProfiler::SegmentProfiler(uint32_t id, bool cupti_disabled)
       output_bytes_(0),
       kernel_profile_state_(ProfilerState::Ready) {}
 
-void SegmentProfiler::startCompile(int device) {
-  device_ = device;
+void SegmentProfiler::startCompile() {
```

Comment:
Separated out set device as a separate function. KernelExecutor knows device on compilation since runtime information is needed for it, the other executors set it in run.